consensus/src/block_storage/sync_manager.rs

```diff
@@ -231,45 +233,68 @@ impl BlockStore {
             + 1;
 
         let blocks = retriever
-            .retrieve_block_for_qc(highest_ordered_cert, num_blocks)
+            .retrieve_block_for_qc(
+                highest_ordered_cert,
+                num_blocks,
+                highest_ledger_info.commit_info().id(),
+            )
             .await?;
 
         assert_eq!(
             blocks.first().expect("should have at least 3-chain").id(),
             highest_ordered_cert.certified_block().id(),
+            "Expecting in the retrieval response, first block should be {}, but got {}",
+            highest_ordered_cert.certified_block().id(),
+            blocks.first().expect("should have at least 3-chain").id(),
```

Comment:
need to update the 3-chain part

Reply:
We meant it. If the blocks ends up empty, we panic with "at least 3-chain". 
I changed it to "blocks are empty" to be more precise.